packages/bemlinter/src/lint.js

```diff
@@ -125,4 +130,13 @@ module.exports = (sources, userOptions = defaultOptions) => {
       }
     });
   }
+  
+  function checkSelector($, filePath, blockName) {
+    eachWrapper($('selector'), wrapper => {
+      const selector = nodeToString(wrapper.node).trim();
+      if (/&[-_a-z]/.test(selector)) {
```

Comment:
I think that, given the pattern you are testing here, you could add a test case to fail on `&foo { //... }`